### PR TITLE
Add support for "deerma.humidifier.mjjsq". Issue: https://github.com/russtone/homebridge-mi-humidifier/issues/19

### PR DESCRIPTION
**Pull request by PietroLubini**
Add support for "deerma.humidifier.mjjsq".
Issue: https://github.com/russtone/homebridge-mi-humidifier/issues/19

### CODE REVIEW

File: devices/MiHumidifierCAL.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const MiHumidifierV1 = require('./MiHumidifierV1');
+
+module.exports = class extends MiHumidifierV1 {
+    constructor() {
+        super();
+        this.version = "cal";
```
Comment (russtone) [inline devices/MiHumidifierCAL.js:6]: Correct version is "ca1":

```suggestion
        this.version = "ca1";
```

Also the name of file in this case should be "MiHumidifierCA1"

File: devices/MiHumidifierCBL.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const MiHumidifierCAL = require('./MiHumidifierCAL');
+
+module.exports = class extends MiHumidifierCAL {
+    constructor() {
+        super();
+        this.version = "cbl";
```
Comment (russtone) [inline devices/MiHumidifierCBL.js:6]: Correct version is "cb1":

```suggestion
        this.version = "cb1";
```

Also the name of file in this case should be "MiHumidifierCB1"

File: devices/MiHumidifierMJJSQL.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    constructor() {
+        super();
+        this.version = "mjjsql";
+    }
+
+    initializePower() {
+        super.initializePower();
+        this.powerGetName = "OnOff_State";
+        this.convertPowerToActivity = function (power) {
+            return power === 1 ? Characteristic.Active.ACTIVE : Characteristic.Active.INACTIVE;
```
Comment (russtone) [inline devices/MiHumidifierMJJSQL.js:14]: Here and in all similar functions `Characteristic` will be undefined.

Comment (russtone) [reply devices/MiHumidifierMJJSQL.js:14]: I've got the following error when tried to start homebridge:

```
getCurrentHumidifierState ReferenceError: Characteristic is not defined
    at module.exports.convertPowerToHumidifierState (/opt/homebridge/npm-global/lib/node_modules/homebridge-mi-humidifier/devices/MiHumidifierV1.js:20:101)
    at MiHumidifier.getCurrentHumidifierState (/opt/homebridge/npm-global/lib/node_modules/homebridge-mi-humidifier/index.js:223:43)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

File: index.js [hunk 4]
```diff
@@ -1,358 +1,451 @@
...
+
+        let options = { ...defaults, ...config };
+        this.infoService = new Service.AccessoryInformation();
+        this.humidifierService = new Service.HumidifierDehumidifier(options.name);
+
+        this.log = log;
+        this.ip = config.ip;
+        this.token = config.token;
+        this.services = [this.humidifierService, this.infoService];
+        this.humidifier = SUPPORTED_HUMIDIFIERS.find(element => element.version === config.model);
```
Comment (russtone) [inline index.js:40]:
```suggestion
        this.humidifier = SUPPORTED_HUMIDIFIERS.find(element => element.version === options.model);
```

Comment (russtone) [reply index.js:40]: Otherwise when there is no `model` in config.json default `v1` won't be applied

File: devices/MiHumidifierCAL.js (hunk 1 above)
Comment (PietroLubini) [reply devices/MiHumidifierCAL.js]: Fixed

File: devices/MiHumidifierCBL.js (hunk 2 above)
Comment (PietroLubini) [reply devices/MiHumidifierCBL.js]: Fixed

File: devices/MiHumidifierMJJSQL.js (hunk 3 above)
Comment (PietroLubini) [reply devices/MiHumidifierMJJSQL.js]: Fixed

File: index.js (hunk 4 above)
Comment (PietroLubini) [reply index.js]: Fixed

File: devices/MiHumidifierV1.js [hunk 5]
```diff
@@ -1,9 +1,11 @@
 const { MODES } = require('constants');
```
Comment (russtone) [inline devices/MiHumidifierV1.js:1]:
```suggestion
const { MODES } = require('./constants');
```

**Pull request merged**